### DISCUSSION TITLE
How COMPOSER_DISABLE_NETWORK=prime should be used?

### DISCUSSION THREAD

**Discussion by vudaltsov** [composer/composer#11719] [Q&A / Support] [answered]
I read https://getcomposer.org/doc/03-cli.md#composer-disable-network and thought that `COMPOSER_DISABLE_NETWORK=prime composer req some/library` can be called before going offline. But for a package that I never downloaded before it says "https://repo.packagist.org could not be fully loaded (Network disabled, request canceled...", although I do have Internet.

I probably got it wrong... What VCS cache priming here means?

**Comment by Seldaek** [comment, marked as answer]
You should rather do a full `composer update` first, then prime if you have vcs repos but that's special use. Then it should work in offline mode.. But really there are very few use cases for this that make sense IMO. What are you trying to achieve?

**Reply by vudaltsov** [reply]
I was on the plane when I noticed that the project I was working on didn't have rector installed. So I ran composer require and it showed me some info about this env variable. I was very glad that with the help of `COMPOSER_DISABLE_NETWORK=1` everything was immediately installed (I had rector in global cache). Later I decided to read more about COMPOSER_DISABLE_NETWORK and came across `prime` and experimented a little. So it's purely out of curiosity.

**Reply by Seldaek** [reply]
Ok, pretty awesome it worked for your case tho, glad to see that :)
